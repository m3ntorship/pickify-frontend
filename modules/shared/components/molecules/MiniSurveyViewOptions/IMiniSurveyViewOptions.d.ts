declare namespace IMiniSurveyViewOptions {
  export interface IProps {
    optionsGroups: IOptionGroups;
  }

  export interface IOption {
    id: string;
    body: string;
  }
  export interface IGroup {
    id: string;
    options: IOption[];
    name: string;
  }

  export interface IOptionGroups {
    groups: IGroup[];
  }
}

export { IMiniSurveyViewOptions };
