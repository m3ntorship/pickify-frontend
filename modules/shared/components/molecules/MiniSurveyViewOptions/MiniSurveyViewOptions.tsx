import React, { useState } from 'react';
import type { FC, ReactElement } from 'react';
import * as EButton from '../../atoms/Button/types/EButton';
import type { IMiniSurveyViewOptions } from './IMiniSurveyViewOptions';
import OptionGroupView from '../OptionGroupView/OptionGroupView';
import Button from '../../atoms/Button/Button';

const MiniSurveyViewOptions: FC<IMiniSurveyViewOptions.IProps> = ({
  optionsGroups,
  addOneVote,
  showExpandButton,
  type,
}): ReactElement => {
  const [isExpanded, setIsExpanded] = useState(false);
  const toggleIsExpanded = (): void => {
    setIsExpanded(!isExpanded);
  };
  return (
    <div className="space-y-4">
      {optionsGroups.groups.map((group) => {
        return (
          <OptionGroupView
            type={type}
            key={group.id}
            group={group}
            addOneVote={addOneVote}
            isExpanded={isExpanded}
          />
        );
      })}
      {showExpandButton ? (
        <Button
          buttonText={
            isExpanded ? 'Collapse all results' : 'Expand all results'
          }
          variant={EButton.buttonVariantValues.TEXT}
          leftIcon
          size={EButton.buttonSizeValues.MEDIUM}
          onClick={toggleIsExpanded}
        />
      ) : (
        ''
      )}
    </div>
  );
};
export default MiniSurveyViewOptions;
