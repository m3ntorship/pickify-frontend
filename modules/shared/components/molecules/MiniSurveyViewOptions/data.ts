export const optionsGroups = {
  groups: [
    {
      id: '03644270-7171-4147-b5a1-4233ff547f7a',
      name: 'Group Name',
      media: [],
      options: [
        {
          id: 'eeb0a32a-1a3c-4c7f-9e50-dd3d3f70b53d',
          body: 'ali tramsy',
          media: [{ url: 'https://placeimg.com/640/480/any' }],
          vote_count: 10,
          voted: false,
        },
        {
          id: 'eeb0a32a-1a3c-4c7f-9e50-dd3d3f70b5aa',
          body: ' ahmed essam',
          media: [{ url: 'https://placeimg.com/640/480/any' }],
          vote_count: 20,
          voted: false,
        },
        {
          id: 'eeb0a32a-1a3c-4c7f-9e50-dd3d3f70b53c',
          body: ' omar gamal',
          media: [{ url: 'https://placeimg.com/640/480/any' }],
          vote_count: 30,
          voted: false,
        },
      ],
    },
  ],
};
