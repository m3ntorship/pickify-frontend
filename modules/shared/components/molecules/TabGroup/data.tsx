import React from 'react';
import { EPostType } from '../../../types/postFeed/EPostType';
import ImagePoll from '../../icons/imagePoll.svg';
import TextPoll from '../../icons/textPoll.svg';
import MiniSurvey from '../../icons/miniSurvey.svg';
import styles from '../../atoms/Tab/Tab.module.css';
import type { ITabGroup } from './ITabGroup';

export const tabGroupData = (): ITabGroup.ITabGroupData[] => {
  return [
    {
      id: '1',
      svg: <ImagePoll className={styles.icon} data-testid="image-poll-icon" />,
      value: 'Image Poll',
      postType: EPostType.ImagePoll,
    },
    {
      id: '2',
      svg: <TextPoll className={styles.icon} />,
      value: 'Text Poll',
      postType: EPostType.TextPoll,
    },
    {
      id: '3',
      svg: <MiniSurvey className={styles.icon} />,
      value: 'Mini Survey',
      postType: EPostType.MiniSurvey,
    },
  ];
};
