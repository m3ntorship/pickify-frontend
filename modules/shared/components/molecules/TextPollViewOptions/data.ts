export const apiDummyData = {
  caption: 'nalyzing Delaware Frozen',
  options_groups: {
    groups: [
      {
        id: '03644270-7171-4147-b5a1-4233ff547f7a',
        name: 'Group Name',
        options: [
          {
            id: 'eeb0a32a-1a3c-4c7f-9e50-dd3d3f70b53d',
            body: 'ali tramsi',
          },
          {
            id: 'eeb0a32a-1a3c-4c7f-9e50-dd3d3f70b5aa',
            body: ' ahmed essam',
          },
          {
            id: 'eeb0a32a-1a3c-4c7f-9e50-dd3d3f70b53c',
            body: ' omar gamal',
          },
        ],
      },
    ],
  },
};
