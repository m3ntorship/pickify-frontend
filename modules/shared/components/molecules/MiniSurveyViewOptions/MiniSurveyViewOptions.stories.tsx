import type { ReactElement } from 'react';
import type { Story } from '@storybook/react';
import MiniSurveyViewOptions from './MiniSurveyViewOptions';
import type { IMiniSurveyViewOptions } from './IMiniSurveyViewOptions';
import { optionsGroups } from './data';

export default {
  title: 'Molecules/MiniSurveyViewOptions',
  component: MiniSurveyViewOptions,
  argTypes: {},
};

const Template: Story<IMiniSurveyViewOptions.IProps> = (args): ReactElement => (
  <MiniSurveyViewOptions {...args} />
);
export const Default = Template.bind({});
Default.args = {
  optionsGroups,
};
