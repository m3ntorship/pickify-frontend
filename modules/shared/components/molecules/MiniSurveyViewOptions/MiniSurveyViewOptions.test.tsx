import * as React from 'react';
import { render, screen } from '@testing-library/react';
import * as renderer from 'react-test-renderer';
import type { TargetElement } from '@testing-library/user-event';
import MiniSurveyViewOptions from './MiniSurveyViewOptions';
import { optionsGroups } from './data';

describe('MiniSurveyViewOptions component', () => {
  it('should render () OptionGroup when we apply ()', () => {
    const tree = renderer
      .create(
        <MiniSurveyViewOptions
          optionsGroups={optionsGroups}
          optionCheckedId="1"
          addOneVote={(): boolean => true}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render 3 Options when we apply options 3 length', () => {
    const numberOfOptions = 3;

    render(
      <MiniSurveyViewOptions
        optionsGroups={optionsGroups}
        optionCheckedId="1"
        addOneVote={(): boolean => true}
      />,
    );

    const option: TargetElement[] = screen.getAllByTestId('option');

    expect(option).toHaveLength(numberOfOptions);
  });
});
