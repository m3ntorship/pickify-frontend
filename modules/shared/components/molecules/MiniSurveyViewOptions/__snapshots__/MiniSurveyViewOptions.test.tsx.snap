// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MiniSurveyViewOptions component should render () OptionGroup when we apply () 1`] = `
<div
  className="space-y-4"
>
  <div
    className="space-y-2"
  >
    <div
      className="text-dark-grey text-sm"
    >
      Group Name
    </div>
    <div
      data-testid="option"
    />
    <div
      data-testid="option"
    />
    <div
      data-testid="option"
    >
      <p>
         omar gamal
      </p>
    </div>
  </div>
  
</div>
`;
